### ISSUE TITLE
[Tune] TypeError: '<' not supported between instances of 'TrackedActor' and 'TrackedActor'

### ISSUE THREAD

**Issue by khaendler** [ray-project/ray#37415]
### What happened + What you expected to happen

Hello all,
running the implementation quickstart example results in the following error:
```
Traceback (most recent call last):
  File "C:\Users\user\miniconda3\lib\site-packages\ray\tune\tune.py", line 1070, in run
    runner.step()
  File "C:\Users\user\miniconda3\lib\site-packages\ray\tune\execution\tune_controller.py", line 284, in step
    self.on_step_end()
  File "C:\Users\user\miniconda3\lib\site-packages\ray\tune\execution\tune_controller.py", line 171, in on_step_end
    self._cleanup_stopping_actors(force_all=False)
  File "C:\Users\user\miniconda3\lib\site-packages\ray\tune\execution\tune_controller.py", line 205, in _cleanup_stopping_actors
    sorted(
TypeError: '<' not supported between instances of 'TrackedActor' and 'TrackedActor'
```

On Linux this error does not occur, only on Windows (both version 10 and 11) with and without installing through miniconda3. This error also always occurs when running similar code to train and evaluate a model on small datasets. On larger datasets this happens on occasion.

After updating pyarrow from 6.0.? to 12.0.1 this error seems to happen only about 50% of the time.
A fix I could find is using time.perf_counter() instead of time.monotonic() for the timestamps in self._stopping_actors in tune_controller.py:

```
    def _remove_actor(self, tracked_actor: TrackedActor):
        stop_future = self._actor_manager.schedule_actor_task(
            tracked_actor, "stop", _return_future=True
        )
        now = time.perf_counter() # initially time.monotonic()
        self._stopping_actors[tracked_actor] = now
        self._earliest_stopping_actor = min(self._earliest_stopping_actor, now)
        self._actor_manager.remove_actor(
            tracked_actor, kill=False, stop_future=stop_future
        )
```

### Versions / Dependencies

ray                            2.5.1
python                     3.10.9
aiosignal                  1.2.0
attrs                         22.1.0
click                         8.1.3
filelock                    3.12.2
frozenlist                 1.3.3
grpcio                     1.51.3
jsonschema            4.17.3
msgpack                 1.0.5
numpy                    1.24.3
packaging               23.0
protobuf                 4.23.3
pyarrow                  12.0.1
pyyaml                    6.0
requests                  2.28.2

### Reproduction script

```
from ray import tune


def objective(config):
    score = config["a"] ** 2 + config["b"]
    return {"score": score}


search_space = {
    "a": tune.grid_search([0.001, 0.01, 0.1, 1.0]),
    "b": tune.choice([1, 2, 3]),
}

tuner = tune.Tuner(objective, param_space=search_space)

results = tuner.fit()
print(results.get_best_result(metric="score", mode="min").config)
```

### Issue Severity

Medium: It is a significant difficulty but I can work around it.

**Comment by xwjiang2010** [comment]
@krfricke Seems related to the new execution logic?

**Comment by krfricke** [comment]
This should be fixed in 2.6.0 (https://github.com/ray-project/ray/commit/443395bf3e97ab0653b6d6000c06d829babbe740#diff-a776b208c6a47c9ed002d89f3d5c3cd35e224f2a54157a23a070ecd014b3abdbR210)
